### DISCUSSION TITLE
Speed Concerns

### DISCUSSION THREAD

**Discussion by gotjoshua** [pdovhomilja/nextcrm-app#43] [General]
I noticed that adding a new section in a project takes really long.
(like 2s for the dialog to close, and 7s for the section to show up)
 Is this just cuz vercel is slow or it there an architectural concern to address?

https://github.com/user-attachments/assets/84da1225-f761-464f-8ee0-e5af717ebb92

**Comment by pdovhomilja** [comment]
Hmm, interesting. Can you send me more info about how it is implemented? Where is it deployed? Where is the hosted DB? We host demo.nextcrm.io on Vercel, and I have never seen this issue before.

**Reply by gotjoshua** [reply]
this is on demo.nextcrm.io actually
https://demo.nextcrm.io/en/projects/boards/6740bb8b9979540efea9a62d
delete is pretty quick, but add has a weird lag. Its not quite as slow tonight as the other night. but still not great.
![Screenshot from 2024-11-27 20-54-38](https://github.com/user-attachments/assets/45f43e6b-a637-42c3-86b5-0b8ecc0eb581)
![Screenshot from 2024-11-27 20-54-34](https://github.com/user-attachments/assets/28d1fd33-eda8-4d76-b1de-fa75b0a5a960)

**Reply by pdovhomilja** [reply]
You are right. I tried the testing on the demo with the same result. But on our OnPrem servers, it is instant. The demo runs on the Vercel PRO plan. I plan to rewrite these API functions to Server actions. I hope this solves the problem.

**Reply by gotjoshua** [reply]
curious how the rewrite will impact it. 

Usually i prefer an optimistic UI update in these scenarios, unless there is big risk for some tx failure or need to roll back.

i didn't try to run locally yet, but if i do, i'll report back.
